### DISCUSSION TITLE
ToolTips use

### DISCUSSION THREAD

**Discussion by matxpa** [2009-Nissan-Cube/About-This-Hack#43] [Ideas]
Hi @0xCUB3 

If you'd like to use ToolTips.

Look at below code (add-ons in ViewController.swift and new component)

ViewController.swift (add-on)
...
    func updateView() {
...
    }
    func setToolTips() {
//       sipToggle.toolTip = sipToggletoolTip
        picture.toolTip = picturetoolTip
//       enrollValue.toolTip = enrollValuetoolTip
        blPrefix.toolTip = blPrefixtoolTip
//        bargsToggle.toolTip = bargsToggletoolTip
//        showHideEnroll.toolTip = showHideEnrolltoolTip
//        changeEnrollment.toolTip = changeEnrollmenttoolTip
        btSysInfo.toolTip = btSysInfotoolTip
        btSoftUpd.toolTip = btSoftUpdtoolTip
    }
...

-------------------------------------------------------------------------------------------

ToolTips.swift (New Swift component)

//
//  ToolTips.swift
//

import Foundation

//let sipToggletoolTip = "Version \(HardwareCollector.OSBuildNum)"
let picturetoolTip = "Icone \(HardwareCollector.OSname) \(HardwareCollector.OSnum)"
//let enrollValuetoolTip = "Inscription au programme de logiciels Regular, Customer Seed, Developer Beta ou Public Beta."
let blPrefixtoolTip = "Bootloader \(HardwareCollector.BootloaderString)"
//let bargsToggletoolTip = "Argument(s) passé(s) au Bootloader\n\(HardwareCollector.BootargsString)"
//let showHideEnrolltoolTip = "Montrer ou cacher l'inscription au programme de logiciels ..."
//let changeEnrollmenttoolTip = "Charger puis modifier l'inscription au programme de logiciels ... (modification devenue impossible depuis Ventura 13.4 beta3 (22F5049e))."
let btSysInfotoolTip = "Informations détaillées du Systeme (matériel et réseau) et des logiciels, logs, historiques ..."
let btSoftUpdtoolTip = "Recherche de mise à jour de \(HardwareCollector.OSname) \(HardwareCollector.OSnum)"

Regards

**Comment by 0xCUB3** [comment]
What would be the purpose? Is it for the edge cases where the text gets cut off?

**Comment by matxpa** [comment]
Hi
for exemple, but not only, it's to get more detailed info

![Capture d’écran 2023-08-12 à 12 45 15](https://github.com/0xCUB3/About-This-Hack/assets/70573409/4952dcb6-39fc-4a60-ab2a-a3ec69094760)


Regards

**Reply by matxpa** [reply]
@0xCUB3  @MDNich

Hi

Finaly I forked ATH and modified components to resolve 
- Add OCLP post-install patch version (issue #24)
- Strange 0.9.1 Release behavior (issue #42)
- OpenCore info is missing after new BootloaderInfo code (issue #48)
and add ToolTip to show OCLP patch info

From this forked ATH pick up what you want and what you need

Regards

**Comment by matxpa** [comment]
@0xCUB3 

May be you'll be interested by one of those 3 DSP RAM Info ToolTips before you publish V1.0.0

![Capture d’écran 2023-09-15 à 18 49 36](https://github.com/0xCUB3/About-This-Hack/assets/70573409/7a1494a5-2e48-4dcb-9150-eaf9c942d64d)

![Capture d’écran 2023-09-15 à 22 04 01](https://github.com/0xCUB3/About-This-Hack/assets/70573409/a3a2bb81-caa1-40f5-8596-5d64dad7aa37)

![Capture d’écran 2023-09-15 à 22 24 26](https://github.com/0xCUB3/About-This-Hack/assets/70573409/3e568203-6a12-4786-aa0e-cc347e2be1dd)

If you want choose one of them
Modifications to do
1) ToolTips.swif (add ramtoolTip)
import Foundation
....
let ramtoolTip = HCRAM.getMemDesc()

2) To add to ViewController.swift (add ram.toolTip)
...
    func setToolTips(){
        ...
        ram.toolTip = ramtoolTip

    }

3) Add that new function in  HCRAM.swift (static func getMemDesc() ) the third exemple and simplest code

    static var thesysmemfile = "~/.ath/sysmem.txt"
...
    static func getMemDesc() -> String {
        return run("echo \"$(egrep \"ECC:|BANK |Size:|Type:|Speed:|Manufacturer:|Part Number:\" " + thesysmemfile + ")\"")
    }

![Capture d’écran 2023-09-16 à 09 27 59](https://github.com/0xCUB3/About-This-Hack/assets/70573409/cc8be247-50a1-4012-a591-feb8ac38eacc)


And new release detection works fine (but Release 1.0.0 is not published, the last one is 0.9.1)

![Capture d’écran 2023-09-16 à 08 53 00](https://github.com/0xCUB3/About-This-Hack/assets/70573409/e46b4d83-78ff-4639-b31f-75daf5fa7639)

Regards

**Reply by 0xCUB3** [reply]
Wow! You never cease to impress me 😀. Now, do you have any knowledge of the issue in #39? I tried rolling back to an older swift version but that did not help.

**Comment by matxpa** [comment]
I’m sorry, I didn’t take a look at my Outlook PM since few days

 

So if you want I stop I’ll stop (lol) 

 

I’ve seen a message where  you asked me why I closed a PR: Because I thought I create it form only one commit, and it show me a PR with 9 or 10 commits, so I deleted it. 

 

De : 0xCUBE ***@***.***>
Répondre à : 0xCUB3/About-This-Hack ***@***.***>
Date : samedi 16 septembre 2023 à 12:17
À : 0xCUB3/About-This-Hack ***@***.***>
Cc : Jean-Michel Meunier ***@***.***>, Author ***@***.***>
Objet : Re: [0xCUB3/About-This-Hack] ToolTips use (Discussion #43)

 

Wow! You never cease to impress me 😀. Now, do you have any knowledge of the issue in #39? I tried rolling back to an older swift version but that did not help.

**Comment by matxpa** [comment]
@0xCUB3 

Hi
You may be interested by this new ATH release 
App and Repo

App        About This Hack.app.zip                       https://drive.google.com/file/d/1TUNAkApff00rp_3yffWo3VUnfozlndnF/view?usp=drive_link
Repo    About-This-Hack-main-matxpa.zip       https://drive.google.com/file/d/1zvnSpJrvDUNjFfmYgt6DR3cCDywvO3XJ/view?usp=drive_link

This repo hasn't been pushed on github



Regards

**Comment by matxpa** [comment]
@0xCUB3  

I apologyze

I've seen some problems, About This Hack.xcodepro is empty ?
in Assets.xcassets Big Sur External or internal SSD or HDD with imagset extension instead of imageset

I've replaced the zipped repo on Google Drive  https://drive.google.com/file/d/16Rh3x7d-p3plsWLlhXo2cUa3SwpqtIqd/view?usp=drive_link

Regards

**Comment by matxpa** [comment]
@0xCUB3

Hi
I desparetly try to upgrade an old Sierra install to High Sierra, it's not easy, what a shame.

Referring to https://github.com/apple/swift/issues/67614

May be a solution ?
Did you try to insert in Reachability.swift a fake variable init. between "class func isConnectedToNetwork() -> Bool {" and "var zeroAddress = sockaddr_in(" 
aka
"        let fakeLocalVar = "This to avoid a Xcode problem :  ATH launch crash on High Sierra"

It's not a dictionary var but ...

Regards

**Comment by matxpa** [comment]
@0xCUB3 @perez987 

Hi
I've just add PCI list at MacModel ToolTip
Only Tooltips.swift component is modified (line 7) (you can see in my Repo (branch main))
      let macModeltoolTip = "\n" + HCMacModel.macName + " - " + HCMacModel.getModelIdentifier() + "\n" 
replaced with
      let macModeltoolTip = "\n" + HCMacModel.macName + " - " + HCMacModel.getModelIdentifier() + "\n\n" + run("system_profiler SPPCIDataType | grep \\":$\\" | sed 's/://g'")

![Capture d’écran 2023-10-06 à 18 12 22](https://github.com/0xCUB3/About-This-Hack/assets/70573409/c7129b75-57c4-4f53-bd06-828dd9c05e3e)

Regards

**Reply by perez987** [reply]
@matxpa 
Amazing!
I’m out of home but I’ll try this asap.
